src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -296,6 +290,33 @@ CreateSingleTaskRouterSelectPlan(DistributedPlan *distributedPlan, Query *origin
 }
 
 
+/*
+ * IsRouterPlannable returns true if the given query can be planned by
+ * router planner.
+ */
+bool
+IsRouterPlannable(Query *query, PlannerRestrictionContext *plannerRestrictionContext)
```

Comment:
We should make sure that we don't get any hard error from the functions called in this function